ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/VertxRequestTransmitter.java

```diff
@@ -12,112 +12,99 @@
 import io.vertx.core.http.HttpClientResponse;
-import io.vertx.core.http.impl.headers.VertxHttpHeaders;
-import io.vertx.ext.web.RoutingContext;
+import io.vertx.core.http.HttpMethod;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-public class VertxRequestTransmitter {
+public class VertxRequestTransmitter implements RequestTransmitter {
 
   private static final Logger LOG = LogManager.getLogger();
+
+  private final Vertx vertx;
   private final Duration httpRequestTimeout;
-  private final ResponseBodyHandler bodyHandler;
+  private final DownstreamResponseHandler bodyHandler;
+  private final HttpClient downStreamConnection;
+  private final DownstreamPathCalculator downstreamPathCalculator;
 
   public VertxRequestTransmitter(
-      final Duration httpRequestTimeout, final ResponseBodyHandler bodyHandler) {
+      final Vertx vertx,
+      final HttpClient downStreamConnection,
+      final Duration httpRequestTimeout,
+      final DownstreamPathCalculator downstreamPathCalculator,
+      final DownstreamResponseHandler bodyHandler) {
+    this.vertx = vertx;
     this.httpRequestTimeout = httpRequestTimeout;
     this.bodyHandler = bodyHandler;
+    this.downStreamConnection = downStreamConnection;
+    this.downstreamPathCalculator = downstreamPathCalculator;
   }
 
-  private void handleException(final RoutingContext context, final Throwable thrown) {
-    if (thrown instanceof TimeoutException || thrown instanceof ConnectException) {
-      context.fail(GATEWAY_TIMEOUT.code(), thrown);
-    } else if (thrown instanceof SSLHandshakeException) {
-      context.fail(BAD_GATEWAY.code(), thrown);
-    } else {
-      context.fail(INTERNAL_SERVER_ERROR.code(), thrown);
-    }
-  }
-
-  public void handleResponse(final RoutingContext context, final HttpClientResponse response) {
-    logResponse(response);
-
-    response.bodyHandler(
-        body ->
-            context
-                .vertx()
-                .executeBlocking(
-                    future -> {
-                      logResponseBody(body);
-                      bodyHandler.handleResponseBody(context, response, body);
-                      future.complete();
-                    },
-                    false,
-                    res -> {
-                      if (res.failed()) {
-                        LOG.error(
-                            "An unhandled error occurred while processing "
-                                + context.getBodyAsString(),
-                            res.cause());
-                        context.fail(res.cause());
-                      }
-                    }));
-  }
-
+  @Override
   public void sendRequest(
-      final HttpClientRequest request, final Buffer bodyContent, final RoutingContext context) {
+      final HttpMethod method,
+      final Map<String, String> headers,
+      final String path,
+      final String body) {
+    LOG.info("Sending a request");
```

Comment:
Probably better at debug level and include the passed in params in the log message

Reply:
done